packages/augur-core/source/contracts/reporting/ShareToken.sol

```diff
@@ -330,8 +335,12 @@ contract ShareToken is ITyped, Initializable, ERC1155, IShareToken, ReentrancyGu
                 logTradingProceedsClaimed(_market, _outcome, _shareHolder, _numberOfShares, _shareHolderShare, _creatorShare.add(_reporterShare));
 
                 if (_proceeds > 0) {
-                    _market.getUniverse().withdraw(address(this), _shareHolderShare.add(_reporterShare), address(_market));
-                    distributeProceeds(_market, _shareHolder, _shareHolderShare, _creatorShare, _reporterShare, _affiliateAddress);
+                    if (toOICash) {
+                        IOICash(_market.getUniverse().getOpenInterestCash()).depositForUser(_shareHolder, _proceeds);
```

Comment:
I'd store OICash on the contract during initialization so we dont have to do all these calls.

Reply:
ah actually nm, market could follow a fork and have a new OICash.